src/syft/core/pointer/pointer.py

```diff
@@ -126,17 +195,23 @@ def request_access(self, request_name: str = "", reason: str = "",) -> None:
         self.location.send_immediate_msg_without_reply(msg=msg)
 
     def check_access(self, node: AbstractNode, request_id: UID) -> any:  # type: ignore
-        from ..node.domain.service import (
-            RequestAnswerMessage,
-            # RequestAnswerResponseService,
-        )
+        """Method that checks the status of an already made request. There are three possible
+        outcomes when requesting access:
+            1. RequestStatus.Accepted - your request has been approved, you can not .get() your
+            data.
+            2. RequestStatus.Pending - your request has not been reviewed yet.
+            3. RequestStatus.Rejected - your request has been rejected.
+
+        :param node: The node that queries the request status.
+        :type node: AbstractNode
+        :param request_id: The request on which you are querying the status.
+        :type request_id: UID
+        """
+        from ..node.domain.service import RequestAnswerMessage
```

Comment:
Q: Can't we add this at the top? or dependency resolution conflict?

Reply:
It's a circular dependency issue at the moment - but it should hopefully be a fixable one.

Reply:
PR welcome - but be wared - here be dragons